presto-hive/src/main/java/io/prestosql/plugin/hive/HiveQlToPrestoTranslator.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive;
+
+import com.google.common.collect.Lists;
+import io.prestosql.spi.PrestoException;
+
+import java.util.Iterator;
+import java.util.ListIterator;
+
+import static io.prestosql.plugin.hive.HiveErrorCode.HIVE_INVALID_VIEW_DATA;
+import static java.lang.String.format;
+
+/**
+ * Translate statements in Hive QL to Presto SQL.
+ *
+ * Only translation of quoted literals is currently included.
+ */
+public final class HiveQlToPrestoTranslator
+{
+    private final String source;
+    // Translation methods consume data from the iterator
+    private final ListIterator<Character> input;
+    private final StringBuilder output = new StringBuilder();
+
+    private HiveQlToPrestoTranslator(String hiveQl)
+    {
+        source = hiveQl;
+        input = Lists.charactersOf(hiveQl).listIterator();
+    }
+
+    /**
+     * Translate a HiveQL statement to Presto SQL by fixing quoted identifiers
+     * and string literals. No other translation is included.
+     *
+     * <p>Backquotes are replaced with double quotes, including SQL-style
+     * escapes (`` becomes ` and " becomes "").
+     *
+     * <p>Single and double quotes are replaced with single quotes, with
+     * minimal processing of escape sequences to ensure that the strings end in
+     * the right place.
+     */
+    public static String translateHiveQuotesToPresto(String hiveStatement)
+    {
+        HiveQlToPrestoTranslator translator =
+                new HiveQlToPrestoTranslator(hiveStatement);
+        return translator.translateQuotedLiterals();
+    }
+
+    private static PrestoException hiveParseError(String message)
+    {
+        // TODO: This error is wrong.
+        return new PrestoException(HIVE_INVALID_VIEW_DATA,
```

Comment:
I don't think you need to split lines here

Reply:
It was 124 characters, but I can make it fit.